internal/metrics/logfilemetricexporter/service_monitor.go

```diff
@@ -0,0 +1,71 @@
+package logfilemetricexporter
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/reconcile"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	prometheusCAFile = "/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt"
+)
+
+func NewServiceMonitor(namespace, name string, owner metav1.OwnerReference) *monitoringv1.ServiceMonitor {
```

Comment:
I see your duplication between components.  Is there a place to locate the common code? Maybe 'internal/metrics' ? since service monitor is metrics related.  I don't know what that would mean for something like "service" though

Reply:
I moved the `service_monitor.go` into `internal/metrics`. For the `service` objects, I moved them into `internal/network` since services are used to expose ports. Not sure if this is the most appropriate.